tensorflow_probability/python/bijectors/frechet_cdf.py

```diff
@@ -0,0 +1,169 @@
+  the Frechet CDF.
+
+  This bijector maps inputs from `[loc, inf]` to `[0, 1]`. The inverse of the
+  bijector applied to a uniform random variable `X ~ U(0, 1)` gives back a
+  random variable with the
+  [Frechet distribution](https://en.wikipedia.org/wiki/Frechet_distribution):
+
+  ```none
+  Y ~ FrechetCDF(loc, scale, concentration)
+  pdf(y; loc, scale, concentration) = (concentration / scale)
+  * exp(- ((x - loc)/scale) ** -concentration)
+  * ((y - loc) / scale) ** - (1 + concentration)
+  ```
+  """
+
+  def __init__(self,
+               loc=0.,
+               scale=1.,
+               concentration=1.,
+               validate_args=False,
+               dtype=tf.float64,
+               name='frechet_cdf'):
+    """Instantiates the `FrechetCDF` bijector.
+
+    Args:
+      loc: Float-like `Tensor` that is the same dtype and is
+        broadcastable with `scale`.
+        This is `loc` in
+        `Y = g(X) = exp(-((X - loc) / scale) ** - concentration)`.
+      scale: Positive Float-like `Tensor` that is the same dtype and is
+        broadcastable with `loc`.
+        This is `scale` in
+        `Y = g(X) = exp(-((X - loc) / scale) ** - concentration)`.
+      concentration: Positive Float-like `Tensor` that is the same dtype and is
+        broadcastable with `loc`.
+        This is `concentration` in
+        `Y = g(X) = exp(-((X - loc) / scale) ** - concentration)`.
+      validate_args: Python `bool` indicating whether arguments should be
+        checked for correctness.
+      name: Python `str` name given to ops managed by this object.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([loc, scale], dtype_hint=dtype)
+      self._loc = tensor_util.convert_nonref_to_tensor(
+          loc, dtype=dtype, name='loc')
+      self._scale = tensor_util.convert_nonref_to_tensor(
+          scale, dtype=dtype, name='scale')
+      self._concentration = tensor_util.convert_nonref_to_tensor(
+          concentration, dtype=dtype, name='concentration')
+      super(FrechetCDF, self).__init__(
+          validate_args=validate_args,
+          forward_min_event_ndims=0,
+          parameters=parameters,
+          dtype=dtype,
+          name=name)
+
+  @property
+  def loc(self):
+    """The `loc` in `Y = exp(-((X - loc) / scale) ** - concentration)`."""
+    return self._loc
+
+  @property
+  def scale(self):
+    """This is `scale` in `Y = exp(-((X - loc) / scale) ** - concentration)`."""
+    return self._scale
+
+  @property
+  def concentration(self):
+    """This is `concentration` in `Y = exp(-((X - loc) / scale) ** - concentration)`."""
+    return self._concentration
+
+  @classmethod
+  def _is_increasing(cls):
+    return True
+
+  def _forward(self, x):
+    with tf.control_dependencies(self._maybe_assert_valid_x(x)):
+      z = (x - self.loc) / self.scale
+      return tf.exp(-z ** (- self.concentration))
```

Comment:
We generally prefer to tf.convert_to_tensor(...) quantities that are used multiple times before using them below. The test I mention will catch this.